### PR TITLE
Anchor links v10

### LINKED ISSUE #513: [Dev]: Anchor link list component
screen shot 2018-12-03 at 9 16 17 am

Anchor links aren't working right now - `gatsby-remark-autolink-headers` because of the custom html we are inserting with remark. Need to fix auto anchor links before creating this component.

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #513

Adds anchor links to all h2 & adds styles first set of links on a page 

### TODO

- [ ] Add to h3 & h4 (causing build errors when I try and duplicate code)

### CODE REVIEW

**Comment by alisonjoseph** [comment]
@asudoh any ideas on why I would get build errors when trying to duplicate the linking code for h3 & h4. 

found this where people were having similar errors but no solution https://github.com/gatsbyjs/gatsby/issues/4208

I've spent all day trying to fix and have decided that for now it might be ok to just have the h2 auto-link, unless you have any ideas? 😓

File: src/components/markdown/Markdown.js [hunk 1]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
   render() {
     return (
-      <div className="ibm--row">
-        <div className="ibm--col-lg-7 ibm--offset-lg-4">
-          <h3 className="page-h3">{this.props.children}</h3>
-        </div>
-      </div>
+      <Location>
+        {({ location }) => {
+          const hash = this.props.children[0].toLowerCase().split(' ').join('-');
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
          const hash = this.props.children[0] && this.props.children[0].toLowerCase().split(' ').join('-');
```

File: src/components/markdown/Markdown.js [hunk 2]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
+          const path = location.pathname;
+          return (
+            <div className="ibm--row">
+              <div className="ibm--col-lg-7 ibm--offset-lg-4">
+                <h3 className="page-h3" id={hash}>     
+                  <Link className="anchor-link" to={`${path}#${hash}`}>
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
                  {hash && (<Link className="anchor-link" to={`${path}#${hash}`}>
```

File: src/components/markdown/Markdown.js [hunk 3]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
+                    <Link20 className="anchor-link__icon" aria-label="Anchor Link" />
+                  </Link>
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
                  </Link>)}
```

File: src/components/markdown/Markdown.js [hunk 4]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
   render() {
     return (
-      <div className="ibm--row">
-        <div className="ibm--col-lg-7 ibm--offset-lg-4">
-          <h3 className="page-h3">{this.props.children}</h3>
-        </div>
-      </div>
+      <Location>
+        {({ location }) => {
+          const hash = this.props.children[0] && this.props.children[0].toLowerCase().split(' ').join('-');
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
          const hash = typeof this.props.children[0] !== 'string' ? undefined : this.props.children[0].toLowerCase().split(' ').join('-');
```

File: src/components/markdown/Markdown.js [hunk 5]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
   render() {
     return (
-      <div className="ibm--row">
-        <div className="ibm--col-lg-7 ibm--offset-lg-4">
-          <h2 className="page-h2">{this.props.children}</h2>
-        </div>
-      </div>
+      <Location>
+        {({ location }) => {
+          const hash = this.props.children[0].toLowerCase().split(' ').join('-');
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
          const hash = typeof this.props.children[0] !== 'string' ? undefined : this.props.children[0].toLowerCase().split(' ').join('-');
```

File: src/components/markdown/Markdown.js [hunk 6]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
+          const path = location.pathname;
+          return (
+            <div className="ibm--row">
+              <div className="ibm--col-lg-7 ibm--offset-lg-4">
+                <h2 className="page-h2" id={hash}>     
+                  <Link className="anchor-link" to={`${path}#${hash}`}>
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
                  {hash && (<Link className="anchor-link" to={`${path}#${hash}`}>
```

File: src/components/markdown/Markdown.js [hunk 7]
```diff
@@ -51,23 +54,49 @@ export class h1 extends React.Component {
...
+                    <Link20 className="anchor-link__icon" aria-label="Anchor Link" />
+                  </Link>
```
Comment (asudoh) [inline src/components/markdown/Markdown.js]:
```suggestion
                  </Link>)}
```

**Pull request merged**